lib/api/CAnomalyJobConfig.cc

```diff
@@ -520,6 +633,290 @@ void CAnomalyJobConfig::CAnalysisConfig::CDetectorConfig::parse(
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_MAX_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationMaxVelocity
+                                  : model::function_t::E_IndividualMaxVelocity;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_MIN_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationMinVelocity
+                                  : model::function_t::E_IndividualMinVelocity;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_MEAN_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationMeanVelocity
+                                  : model::function_t::E_IndividualMeanVelocity;
+        argumentRequired = true;
+    } else if (m_FunctionName == FUNCTION_SUM_VELOCITY) {
+        m_Function = isPopulation ? model::function_t::E_PopulationSumVelocity
+                                  : model::function_t::E_IndividualSumVelocity;
+        argumentRequired = true;
+    } else {
+        if (haveSummaryCountField) {
+            LOG_ERROR(<< "Implicit function metric() cannot be "
+                         "used with a summary count field");
+            return false;
+        }
+
+        m_Function = isPopulation ? model::function_t::E_PopulationMetric
+                                  : model::function_t::E_IndividualMetric;
+
+        // This is inconsistent notation, but kept for backwards compatibility
+        m_FieldName = m_FunctionName;
+
+        return true;
+    }
+
+    // Validate
+    if (model::function_t::isPopulation(m_Function) && isPopulation == false) {
+        LOG_ERROR(<< "Function " << m_FunctionName << "() requires an 'over' field");
+        return false;
+    }
+
+    if (isPopulation && model::function_t::isPopulation(m_Function) == false) {
+        LOG_ERROR(<< "Function " << m_FunctionName << "() cannot be used with an 'over' field");
+        return false;
+    }
+
+    if (byFieldRequired && hasByField == false) {
+        LOG_ERROR(<< "Function " << m_FunctionName << "() requires a 'by' field");
+        return false;
+    }
+
+    if (byFieldInvalid && hasByField) {
+        LOG_ERROR(<< "Function " << m_FunctionName << "() cannot be used with a 'by' field");
+        return false;
+    }
+
+    if (argumentRequired && m_FieldName.empty()) {
+        LOG_ERROR(<< "Function " << m_FunctionName << "() requires an argument");
+        return false;
+    }
+
+    if (argumentInvalid && m_FieldName.empty() == false) {
+        LOG_ERROR(<< "Function " << m_FunctionName << "() does not take an argument");
+        return false;
+    }
+
+    return true;
+}
+
+bool CAnomalyJobConfig::CAnalysisConfig::CDetectorConfig::decipherExcludeFrequentSetting() {
+
+    bool hasByField{m_ByFieldName.empty() == false};
+    bool isPopulation{m_OverFieldName.empty() == false};
+
+    if (m_ExcludeFrequent.empty() == false) {
```

Comment:
Is there a unit test that tests this case?  Most seem to be testing the path where `m_ExcludeFrequent.empty() == true`.